enclave-manager/web/src/client/enclaveManager/websocketClient/WebSocketClient.d.ts

```diff
@@ -0,0 +1,43 @@
+import type {
+    ErrorResponse,
+    SuccessResponse,
+    FilterKeys,
+    MediaType,
+    PathsWithMethod,
+    ResponseObjectMap,
+    OperationRequestBodyContent,
+    HasRequiredKeys,
+} from "openapi-typescript-helpers";
+
+import type {
+    FetchResponse,
+    ParamsOption,
+    QuerySerializer
+} from "openapi-fetch"
+
+export interface ClientOptions {
+    baseUrl?: string;
+}
+
+export type RequestOptions<T> = ParamsOption<T> & {
+    querySerializer?: QuerySerializer<T>;
+    abortSignal?: AbortSignal;
+};
+
+// This implementation is based on the http version of the lib `openapi-fetch`
+// https://github.com/drwpow/openapi-typescript/blob/main/packages/openapi-fetch/src/index.d.ts
+export default function createWSClient<Paths extends {}>(
+    clientOptions?: ClientOptions,
+): {
+    /** Call a WS endpoint */
+    WS<P extends PathsWithMethod<Paths, "get">>(
+        url: P,
+        ...init: HasRequiredKeys<
+            RequestOptions<FilterKeys<Paths[P], "get">>
+        > extends never
+            ? [(RequestOptions<FilterKeys<Paths[P], "get">> | undefined)?]
+            : [RequestOptions<FilterKeys<Paths[P], "get">>]
+    ): AsyncGenerator<FetchResponse<"get" extends infer T ? T extends "get" ? T extends keyof Paths[P] ? Paths[P][T] : unknown : never : never>>;
+};
+
+export { };
```

Comment:
We should avoid using type definition files alongside JS as much as possible imo:
* There is a higher developer maintenance burden because two files must now be maintained for a single piece of functionality.
* Inconsistencies between (complex) types and implementation are generally only going to be discovered at runtime.

This client could be written as a single `.ts` file right?

Reply:
Agreed

Reply:
Agreed! Not sure why `openapi-fetch` went with JS + type interface but I don't a strong reason make into a pure TS  implementation and have all the benefits of the type system.

Reply:
done :D